cpp/include/raft/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -40,6 +42,23 @@
 
 namespace raft::neighbors::cagra::detail {
 
+template <typename DataT, typename accessor>
+ivf_pq::index_params get_default_ivf_pq_build_params(
+  mdspan<const DataT, matrix_extent<int64_t>, row_major, accessor> dataset,
+  const raft::distance::DistanceType metric)
+{
+  auto build_params    = ivf_pq::index_params{};
+  build_params.n_lists = dataset.extent(0) < 4 * 2500 ? 4 : (uint32_t)(dataset.extent(0) / 2500);
```

Comment:
We've been doing some studies with these params and have found some more reasonable defaults than what we've been using. Given that we're changing these in this PR anyways, I think we have an opportunity to update these here. 

We should update this to be sqrt(n_vectors)